### DISCUSSION TITLE
Question on one sample

### DISCUSSION THREAD

**Discussion by hebersarmiento** [spakin/SimpInkScr#81] [Q&A] [answered]
Greetings,

I am happy with this extension, and I have worked on few function definitions of my own. If you do not mind, could you please help me with the following?

a=path [...]
obj = a.get_inkex_object()
obj.set('d', obj.path.rotate(-90, inkex.Vector2d(x+22,y)))

I would like to have the correspondent of the code above for translations, using the same structure (using inkex).

Any help would be highly appreciated.

P.S.

I have tried using
a.translate() and a.rotate()
but I cannot append 'a' as it displays an error message indicating it is not a path.

**Comment by spakin** [comment, marked as answer]
I'm not sure I understand your question.  Are you asking how to translate the path coordinates?  If so, then you should be able to invoke `translate` on the `inkex` path object:
```Python
obj.set('d', obj.path.rotate(-90, inkex.Vector2d(x+22,y)).translate(500, 500))
```
If that doesn't answer your question, could you please provide more information about what you're trying to do?

**Reply by hebersarmiento** [reply]
I just needed the syntax to do translation. Then you hand me both translation and rotation in one shot! You have answered my question. I was trying to create functions to draw few electric components and offer options to translate/rotate them. So basically I draw several paths and connect them using the append method. 
    c = path(['M', x, y, 'H', x + 15, 'V',y+6,'V',y-6], stroke_width=.5)
    obj = c.get_inkex_object()
    obj.set('d', obj.path.rotate(0, inkex.Vector2d(x, y)).translate(0, 0)) #this will perform translation and rotation but change the 0's
    c.append(obj)
    ... (more lines)
Thank you so much for the prompt response, and I would like to ask you (if you prefer me to write a different question I will) what the **'d'** means in the line:
_obj.set(**'d'**, obj.path.rotate(0, inkex.Vector2d(x, y)).translate(0, 0))_

**Reply by spakin** [reply]
> Thank you so much for the prompt response, and I would like to ask you (if you prefer me to write a different question I will) what the 'd' means in the line:

The `d` attribute specifies a string of path comments, as in `d = "M 100 100 H 115 V 106 V 94"`.  See the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d) for full details.  Your code is transforming the path itself by applying rotations, translations, etc. to those path commands, replacing the original `d` string with a modified one.

Inspired by your question, I just committed 0836ca58d4c232324b906547c9965341c66e331b, which provides the same path-transformation functionality directly on Simple Inkscape Scripting path objects (i.e., on `a` as opposed to `obj` in your original message).  See [Transforming a path](https://github.com/spakin/SimpInkScr/wiki/Modifying-existing-objects#transforming-a-path) for documentation.  In short, one can now write
```Python
a=path [...]
a.rotate_path(-90, (x+22,y))
```
instead of
```Python
a=path [...]
obj = a.get_inkex_object()
obj.set('d', obj.path.rotate(-90, inkex.Vector2d(x+22,y)))
```
